File: backend/src/main/kotlin/com/pwarss/model/NewsEntry.kt
// Created by Konstantin Khvan on 7/6/18 5:21 PM

package com.pwarss.model

import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer
import java.sql.Date

/**
 * News entry dto
 */
data class NewsEntry(@JsonSerialize(using = ToStringSerializer::class) val id: Long,
                     val tags: List<String>,
                     val link: String,
                     val title: String,
                     val date: Date,
                     val content: String)

/**
 * Null object for NewsEntry
 */
val EMPTY_ENTRY = NewsEntry(1, emptyList(), "", "", Date(0), "")


val TAG_READ = "read"
val TAG_MARKED = "marked"

File: frontend/src/model/NewsEntry.ts
// Created by Konstantin Khvan on 7/16/18 9:24 AM


export type IdType = string

export class NewsEntry {
    id: IdType = "";
    tags: string[] = [];
    link: string = "";
    title: string = "";
    date: string = "";
    content: string = "";
}

export const NullEntry: NewsEntry = {
    id: "-1",
    tags: [],
    link: "",
    title: "",
    date: "",
    content: ""
};


export const TAG_READ = "read"
export const TAG_MARKED = "marked"

export function newsEntryStarred(entry: NewsEntry): boolean {
    return !!entry.tags && (entry.tags.indexOf(TAG_MARKED) != -1);
}

export function newsEntryRead(entry: NewsEntry): boolean {
    return !!entry.tags && (entry.tags.indexOf(TAG_READ) != -1);
}
